repos/system_upgrade/el7toel8/actors/selinux/selinuxapplycustom/actor.py

```diff
@@ -0,0 +1,102 @@
+                self.log.info("Installing module %s on priority %d.", module.name, module.priority)
+                if module.removed:
+                    self.log.warning("The following lines where removed because of incompatibility: \n%s",
+                                     '\n'.join(module.removed))
+                # write module content to disk
+                try:
+                    with open(cil_filename, 'w') as cil_file:
+                        cil_file.write(module.content)
+                except OSError as e:
+                    self.log.warning("Error writing %s : %s", cil_filename, str(e))
+                    continue
+
+                try:
+                    run(['semodule',
+                         '-X', str(module.priority),
+                         '-i', cil_filename
+                         ]
+                        )
+                except CalledProcessError as e:
+                    self.log.warning("Error installing module: %s", str(e.stderr))
+                    # TODO - save the failed module to /etc/selinux ?
+                    # currently it is still left in the old policy store
+                try:
+                    os.remove(cil_filename)
+                except OSError as e:
+                    self.log.warning("Error removing module file: %s", str(e))
+        # import SELinux customizations collected by "semanage export"
+        for custom in self.consume(SELinuxCustom):
+            self.log.info('Importing the following SELinux customizations collected by "semanage export": \n%s',
+                          '\n'.join(custom.commands))
+            semanage_filename = os.path.join(WORKING_DIRECTORY, "semanage")
+            # save SELinux customizations to disk
+            try:
+                with open(semanage_filename, 'w') as s_file:
+                    s_file.write('\n'.join(custom.commands))
+            except OSError as e:
+                self.log.warning("Error writing SELinux customizations: %s", str(e))
+            # import customizations
+            try:
+                run(['semanage', 'import', '-f', semanage_filename])
+            except CalledProcessError as e:
+                self.log.warning("Failed to import SELinux customizations: %s", str(e.stderr))
+                continue
+            # clean-up
+            try:
+                os.remove(semanage_filename)
+            except OSError as e:
+                self.log.warning("Failed to remove temporary file %s: %s", semanage_filename, str(e))
+                continue
+
+        # clean-up
+        rmtree(WORKING_DIRECTORY, ignore_errors=True)
+
+        # TODO - Verify that all RPM packages reqested by selinux actors are installed
+        self.log.info("Verifying selinux-related RPMs requested before upgrade.")
+        for rpms in self.consume(SELinuxRequestRPMs):
+            self.log.info("To keep: %s \n To install: %s", ", ".join(rpms.to_keep), ", ".join(rpms.to_install))
```

Comment:
Sorry if I'm missing it, but where is the verification that requested rpms have been installed? I'm a bit mislead by the TODO comment - is it part of this patch or a follow up?

Reply:
It's not implemented yet. I don't know how. Back when I wrote the comment I was told it would be somehow possible to check installed RPM's. But at this point I'm fine with "best effort".